workflow/controller/retry_tweak.go

```diff
@@ -14,6 +14,13 @@ type RetryTweak = func(retryStrategy wfv1.RetryStrategy, nodes wfv1.Nodes, pod *
 // FindRetryNode locates the closes retry node ancestor to nodeID
 func FindRetryNode(nodes wfv1.Nodes, nodeID string) *wfv1.NodeStatus {
 	boundaryID := nodes[nodeID].BoundaryID
+	if boundaryID == "" {
+		for _, node := range nodes {
```

Comment:
This is yet another loop. Can we somehow combine it with the loop below?

Reply:
modifid, please take a look, thanks!